data-prepper-plugins/rss-source/src/test/java/org/opensearch/dataprepper/plugins/source/rss/RSSSourceTest.java

```diff
@@ -32,54 +33,35 @@ class RSSSourceTest {
 
     private final String VALID_RSS_URL = "https://forum.opensearch.org/latest.rss";
 
-    private final String INVALID_RSS_URL = "https://docs.aws.amazon.com/amazondynamodb/latest";
-
-    private PluginMetrics pluginMetrics;
+    @Mock
+    private Buffer<Record<Document>> buffer;
 
+    @Mock
     private RSSSourceConfig rssSourceConfig;
 
+    private PluginMetrics pluginMetrics;
+
     private RSSSource rssSource;
 
     @BeforeEach
     void setUp() {
         pluginMetrics = PluginMetrics.fromNames(PLUGIN_NAME, PIPELINE_NAME);
-        rssSourceConfig = mock(RSSSourceConfig.class);
-        rssSource = mock(RSSSource.class);
-    }
-
-    @Test
-    void test_when_extractItemsFromRssFeed_captured() {
-        ArgumentCaptor<RSSSourceConfig> argumentCaptor = ArgumentCaptor.forClass(RSSSourceConfig.class);
-        when(rssSourceConfig.getUrl()).thenReturn(VALID_RSS_URL);
-        doNothing().when(rssSource).extractItemsFromRssFeed(argumentCaptor.capture());
-        rssSource.extractItemsFromRssFeed(rssSourceConfig);
-        assertEquals(rssSourceConfig, argumentCaptor.getValue());
-    }
-
-    @ParameterizedTest
-    @ValueSource(strings = { "https://www.yahoo.com/news/rss/mostviewed", "https://forum.opensearch.org/latest.rss",
```

Comment:
You might consider adding some of these old tests to a new test suite for integration tests: `RssSourceIT`. I think this could be a different PR.

Reply:
Yes that's the plan. I removed these from unit test and intend to add them as integration tests in a future PR.